dask_ml/feature_extraction/text.py

```diff
@@ -108,3 +112,253 @@ class FeatureHasher(_BaseHasher, sklearn.feature_extraction.text.FeatureHasher):
+        client = get_client()
+        params = self.get_params()
+        vocabulary = params.pop("vocabulary")
+        del params["use_actors"]
+
+        vocabulary_ = client.submit(Vocabulary, vocabulary=vocabulary, actor=True)
+        vocabulary_actor = vocabulary_.result()
+
+        if self.vocabulary is not None:
+            # Case 1: Just map transform.
+            fixed_vocabulary = True
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform, vocabulary_actor, params
+            )
+        else:
+            fixed_vocabulary = False
+            # Case 2: learn vocabulary from the data.
+            vocabularies = raw_documents.map_partitions(
+                _build_vocabulary, vocabulary_actor, params
+            )
+            # compute and merge.
+            dask.compute(vocabularies)  # List[Set[str]]
+            # same as before
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform, vocabulary_actor, params
+            )
+
+        meta = scipy.sparse.eye(0, format="csr", dtype=self.dtype)
+        result = build_array(result, vocabulary_actor.n_features, meta)
+
+        self.vocabulary_actor_ = vocabulary_actor
+        self.vocabulary_ = vocabulary_actor.finalize().result()
+        self.fixed_vocabulary_ = fixed_vocabulary
+
+        return result
+
+    def transform(self, raw_documents):
+        if self.use_actors:
+            return self._transform(raw_documents)
+        else:
+            return self._transform_no_actor(raw_documents)
+
+    def _transform(self, raw_documents):
+        params = self.get_params()
+        del params["vocabulary"]
+        del params["use_actors"]
+
+        check_is_fitted(self, "vocabulary_")
+        result = raw_documents.map_partitions(
+            _count_vectorizer_transform, self.vocabulary_actor_, params
+        )
+        meta = scipy.sparse.eye(0, format="csr", dtype=self.dtype)
+        return build_array(result, self.vocabulary_actor_.n_features, meta)
+
+    def _fit_transform_no_actor(self, raw_documents, y=None):
+        # Just bag for now.
+        # Two cases:
+        # 1. vocabulary provided, easy.
+        # 2. vocabulary learned, harder.
+        params = self.get_params()
+        vocabulary = params.pop("vocabulary")
+        del params["use_actors"]
+
+        if self.vocabulary is not None:
+            # Case 1: Just map transform.
+            fixed_vocabulary = True
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform_no_actor, dask.delayed(vocabulary), params
+            )
+        else:
+            fixed_vocabulary = False
+            # Case 2: learn vocabulary from the data.
+            vocabularies = raw_documents.map_partitions(
+                _build_vocabulary_no_actor, params
+            )
+
+            # compute
+            vocabulary = _merge_vocabulary(*vocabularies.to_delayed()).compute()
```

Comment:
Do we need to bring this local and call compute, or can we delay the `_merge_vocabulary` call?  It looks like we might need to bring this locally eventually, but hopefully we can do all of this computation at once, rather than pull things locally, and then push them back and start computation again.

Reply:
Still futzing with this, but I think it's unavoidable.

1. We need the length to get the output shape right (number of columns)
2. We need the concrete result eventually to provide to the user as vocabulary_

Reply:
Right, but we don't need to pull down and then push back up.  Rather, I recommend that we keep everything delayed, and then when we're done we pull things down rather than re-scatter them back up.  This makes it so that this can be run locally without the distributed scheduler, and it reduces communication back and forth.

```suggestion
            remote_vocabulary = _merge_vocabulary(*vocabularies.to_delayed())
```

Reply:
Ah, I see we don't call compute after this.

Maybe then we do

```python
remote_vocabulary = remote_vocabulary.persist()  # compute, and keep in remote memory
local_vocabulary = remote_vocabulary.compute()
```